src/main/java/org/jenkinsci/remoting/util/KeyUtils.java

```diff
@@ -96,6 +98,7 @@ private static boolean equals(String str1, String str2) {
      * @return the MD5 fingerprint of the key.
      */
     @Nonnull
+    @SuppressFBWarnings(value = "WEAK_MESSAGE_DIGEST_MD5", justification = "Used for fingerprinting, not security.")
```

Comment:
You never know, because it is a utility class

Reply:
Yes, we really ought to clean this up, replace some of these old algorithms. I'd like to see if we can get some of that going sometime.